lib/pages/chat_list/chat_list_item.dart

```diff
@@ -201,142 +202,215 @@ class ChatListItem extends StatelessWidget {
+              ],
+            ),
+          ),
+          onTap: () => clickAction(context),
+        ),
+      ),
+    );
+  }
+
+  FutureBuilder<String> textContentWidget(BuildContext context, bool isGroup) {
+    return FutureBuilder<String>(
+      future: room.lastEvent?.calcLocalizedBody(
+            MatrixLocals(L10n.of(context)!),
+            hideReply: true,
+            hideEdit: true,
+            plaintextBody: true,
+            removeMarkdown: true,
+          ) ??
+          Future.value(L10n.of(context)!.emptyChat),
+      builder: (context, snapshot) {
+        return Text(
+          room.membership == Membership.invite
+              ? L10n.of(context)!.youAreInvitedToThisChat
+              : snapshot.data ??
+                  room.lastEvent?.calcLocalizedBodyFallback(
+                    MatrixLocals(L10n.of(context)!),
+                    hideReply: true,
+                    hideEdit: true,
+                    plaintextBody: true,
+                    removeMarkdown: true,
+                  ) ??
+                  L10n.of(context)!.emptyChat,
+          softWrap: false,
+          maxLines: isGroup ? 1 : 2,
+          overflow: TextOverflow.ellipsis,
+          style: Theme.of(context).textTheme.bodySmall?.copyWith(letterSpacing: 0.4),
+        );
+      },
+    );
+  }
+
+  Row typingTextWidget(String typingText, BuildContext context) {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.center,
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: [
+        Flexible(
+          child: Text(
+              typingText,
+              style: Theme.of(context).textTheme.labelLarge?.merge(
+                    TextStyle(
+                      overflow: TextOverflow.ellipsis,
+                      letterSpacing: -0.007,
+                      fontWeight: FontWeight.w400,
+                      color: Theme.of(context).colorScheme.primary,
+                      fontSize: ChatListItemStyle.isTypingFontSize,
                     ),
                   ),
-                ],
-              ),
-              onTap: () => clickAction(context),
+              maxLines: 1,
+              softWrap: true,
             ),
-          ],
         ),
-      ),
+        const SizedBox(width: 5.25),
+        const TypingIndicator(showIndicator: true)
+      ],
     );
   }
+
+  RenderObjectWidget lastSenderWidget(bool isGroup, bool unread) {
+    return isGroup
+        ? Row(
+            children: [
+              FutureBuilder<User?>(
+                future: room.lastEvent?.fetchSenderUser(),
+                builder: (context, snapshot) {
+                  if (snapshot.data == null) return const SizedBox.shrink();
+                  return Text(
+                    snapshot.data!.displayName!,
+                    overflow: TextOverflow.ellipsis,
+                    maxLines: 1,
+                    softWrap: false,
+                    style: Theme.of(context).textTheme.titleMedium?.merge(
```

Comment:
IMO, it is label large

<img width="626" alt="image" src="https://github.com/linagora/twake-on-matrix/assets/6462404/7ced52a6-39ad-4ab6-a19a-3dbd766e2d1b">